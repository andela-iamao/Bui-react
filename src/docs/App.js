import React from 'react';
import PropTypes from 'prop-types';

import NotFound from './NotFound';
import Component from './Component';
import Navigation from './Navigation';

import componentData from '../../config/componentData';


const PATH = window.location.pathname.substr(1);


class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = { route: props.location || PATH };
    }

    render() {
        const { route } = this.state;

        const componentMap = componentData.reduce((a, component) => {
            a[component.name.toLowerCase()] = component;
            return a;
        }, {});

        const component = route ?
            componentMap[route.toLowerCase()]
            :
            componentMap[componentData[0].name.toLowerCase()];
        
        if (!component) {
            return <NotFound />;
        }

        return (
            <div className="row">
                <div className="col-md-2" id="navigation-container">
                    <Navigation
                      path={PATH.toLowerCase()}
                      components={componentData.map((component) => component.name)}
                    />
                </div>
                <div className="col-md-8" id="details-container">
                    <Component component={component}/>
                </div>
            </div>
        )
    }
}

App.propTypes = {
    location: PropTypes.string
};

export default App;
