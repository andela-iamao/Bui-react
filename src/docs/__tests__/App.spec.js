import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { mount } from 'enzyme';

import App from '../App';


describe('DOCS: App', () => {

    describe('Snapshot Test', () => {
        it('renders the component page if a valid route is found', () => {
            const tree = renderer.create(<App />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders 404 page if a valid route is not found', () => {
            const tree = renderer.create(<App location="lol" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('Integration Test', () => {
        it('renders documentation and examples', () => {
            const div = document.createElement('div');
            ReactDOM.render(<App />, div);
            ReactDOM.unmountComponentAtNode(div);
        });

        it('renders not found page on non-existent components', () => {
            const wrapper = mount(<App location='abc' />);

            expect(wrapper.find({ id: 'bui-react-docs-component-not-found' })).toHaveLength(1);
            expect(wrapper.find({ id: 'bui-react-docs-component-not-found' }).children().text()).toEqual('404!');
        });
    });
});
