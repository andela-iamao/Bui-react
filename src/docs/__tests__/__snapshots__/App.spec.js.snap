// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOCS: App Snapshot Test renders 404 page if a valid route is not found 1`] = `
<div
  id="bui-react-docs-component-not-found"
>
  <h3>
    404!
  </h3>
</div>
`;

exports[`DOCS: App Snapshot Test renders the component page if a valid route is found 1`] = `
<div
  className="row"
>
  <div
    className="col-md-2"
    id="navigation-container"
  >
    <div>
      <div
        id="navigation-logo"
      >
        <h4>
          bui React
        </h4>
      </div>
      <div
        id="navigation-items"
      >
        <ul
          className="navigation"
        >
          <li
            className="bui-react-docs-component-name active"
          >
            <a
              href="/"
            >
              Getting started
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              href="/button"
            >
              Button
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="col-md-8"
    id="details-container"
  >
    <div
      id="bui-react-docs-component-container"
    >
      <h1
        id="bui-react-docs-component-name"
      >
        Button
      </h1>
      <p
        id="bui-react-docs-component-description"
      >
        Button Component
      </p>
      <br />
      <br />
      <h3
        id="bui-react-docs-component-example-header"
      >
        Example
      </h3>
      <div
        id="bui-react-docs-component-example-container"
      >
        <h4
          id="bui-react-docs-component-example-description"
        >
          Default Button component
        </h4>
        <button
          data-radium={true}
          style={
            Object {
              "backgroundColor": "#FFF",
              "border": "1px solid #333",
              "color": "#333",
              "minWidth": 150,
              "padding": 10,
            }
          }
        >
          Hello!
        </button>
        <p
          id="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
      </div>
      <br />
      <br />
      <h3
        id="bui-react-docs-component-props-header"
      >
        Props
      </h3>
      <div
        id="bui-react-docs-component-props-container"
      >
        <table
          className="props"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Type
              </th>
              <th>
                Default
              </th>
              <th>
                Required
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                children
              </td>
              <td>
                Child elements for the button component
              </td>
              <td>
                node
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                onClick
              </td>
              <td>
                
              </td>
              <td>
                func
              </td>
              <td />
              <td />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
